File: tarefas/script/conexao.cpp
#include <iostream>
#include <pqxx/pqxx>

int main() {
    try {
        // Estabelecer conexão com o banco de dados
        pqxx::connection conn("dbname=postgres user=postgres password=postgres hostaddr=127.0.0.1 port=5432");

        if (conn.is_open()) {
            std::cout << "Conexão bem-sucedida!" << std::endl;
        } else {
            std::cout << "Erro ao conectar." << std::endl;
            return 1;
        }

        // Executar consulta SQL
        pqxx::work txn(conn);
        pqxx::result result = txn.exec("SELECT * FROM projeto");

        // Exibir resultados
        for (const auto &row : result) {
            for (const auto &field : row) {
                std::cout << field.c_str() << " ";
            }
            std::cout << std::endl;
        }

        // Commit e fechamento da transação
        txn.commit();

        // Fechar conexão
        conn.disconnect();
        std::cout << "Conexão encerrada." << std::endl;
    } catch (const std::exception &e) {
        std::cerr << "Erro ao conectar ao PostgreSQL: " << e.what() << std::endl;
        return 1;
    }

    return 0;
}

File: tarefas/script/conexao.py
from sqlalchemy import create_engine, MetaData, Table
from sqlalchemy.sql import text

# Conectar ao banco de dados PostgreSQL
engine = create_engine('postgresql://postgres:postgres@localhost:5432/postgres')
connection = engine.connect()

# Função para obter consulta SQL do usuário
def get_user_sql():
    sql = input("Insira a consulta SQL: ")
    return sql

# Executar consulta SQL inserida pelo usuário
def execute_sql(sql):
    result = connection.execute(text(sql))
    return result

try:
    # Loop para permitir múltiplas consultas
    while True:
        user_sql = get_user_sql()
        if user_sql.lower() == "exit":
            break  # Saia do loop se o usuário digitar 'exit'
        result = execute_sql(user_sql)

        # Exibir resultados
        for row in result:
            print(row)

except KeyboardInterrupt:
    print("\nOperação interrompida pelo usuário.")

finally:
    # Fechar conexão ao final
    connection.close()
